spark-connector/src/main/java/com/datastrato/gravitino/spark/connector/iceberg/IcebergAdaptor.java

```diff
@@ -0,0 +1,146 @@
+        "Couldn't get "
+            + GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_WAREHOUSE
+            + " from iceberg catalog properties");
+    icebergProperties.put(
+        GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_TYPE,
+        catalogBackend.toLowerCase(Locale.ENGLISH));
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_URI, metastoreUri);
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_WAREHOUSE, hiveWarehouse);
+  }
+
+  private void initJdbcProperties(
+      String catalogBackend,
+      Map<String, String> gravitinoProperties,
+      HashMap<String, String> icebergProperties) {
+    String jdbcUri = gravitinoProperties.get(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_URI);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jdbcUri),
+        "Couldn't get "
+            + GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_URI
+            + " from iceberg catalog properties");
+    String jdbcWarehouse =
+        gravitinoProperties.get(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_WAREHOUSE);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jdbcWarehouse),
+        "Couldn't get "
+            + GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_WAREHOUSE
+            + " from iceberg catalog properties");
+    String jdbcUser = gravitinoProperties.get(GravitinoSparkConfig.GRAVITINO_JDBC_USER);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jdbcUser),
+        "Couldn't get "
+            + GravitinoSparkConfig.GRAVITINO_JDBC_USER
+            + " from iceberg catalog properties");
+    String jdbcPasswrod = gravitinoProperties.get(GravitinoSparkConfig.GRAVITINO_JDBC_PASSWORD);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jdbcPasswrod),
+        "Couldn't get "
+            + GravitinoSparkConfig.GRAVITINO_JDBC_PASSWORD
+            + " from iceberg catalog properties");
+    String jdbcDriver =
+        gravitinoProperties.get(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_DRIVER);
+    Preconditions.checkArgument(
+        StringUtils.isNotBlank(jdbcDriver),
+        "Couldn't get "
+            + GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_DRIVER
+            + " from iceberg catalog properties");
+    icebergProperties.put(
+        GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_TYPE,
+        catalogBackend.toLowerCase(Locale.ENGLISH));
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_URI, jdbcUri);
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_WAREHOUSE, jdbcWarehouse);
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_USER, jdbcUser);
+    icebergProperties.put(
+        GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_PASSWORD, jdbcPasswrod);
+    icebergProperties.put(GravitinoSparkConfig.LAKEHOUSE_ICEBERG_CATALOG_JDBC_DRIVER, jdbcDriver);
+    icebergProperties.put(
```

Comment:
seems we could remove `LAKEHOUSE_ICEBERG_CATALOG_JDBC_INITIALIZE `, because it's used to init jdbc tables when loading catalog which is done in Gravitino

Reply:
updated.